tests/test_silicon_platform_amaranth.py

```diff
@@ -0,0 +1,574 @@
+        platform_port = SiliconPlatformPort("comp", "test_input", port_obj)
+
+        # Create buffer
+        buffer = IOBuffer("i", platform_port)
+
+        # Create module and elaborate
+        m = Module()
+        m.submodules.buffer = buffer
+
+        # Get the fragment
+        fragment = Fragment.get(m, None)
+
+        # Just check that elaboration succeeds without error
+        self.assertIsNotNone(fragment)
+
+    def test_elaborate_o(self):
+        """Test IOBuffer elaborate with output port"""
+        # Create a simple test platform
+        class TestPlatform:
+            pass
+
+        # Create an output port
+        port_obj = Port(type="output", pins=["1", "2"], port_name="test_output",
+                       direction=io.Direction.Output, options={})
+        platform_port = SiliconPlatformPort("comp", "test_output", port_obj)
+
+        # Create buffer with the proper signals
+        buffer = IOBuffer("o", platform_port)
+        # Explicitly set buffer signals to match what would be set in actual use
+        buffer.o = Signal(2, name="output_signal")
+        buffer.oe = Signal(1, name="enable_signal", init=-1)  # Output enabled by default
+
+        # Create module and elaborate
+        m = Module()
+        m.submodules.buffer = buffer
+
+        # Get the fragment
+        fragment = Fragment.get(m, TestPlatform())
+
+        # Just check that elaboration succeeds without error
+        self.assertIsNotNone(fragment)
+
+    def test_elaborate_io(self):
+        """Test IOBuffer elaborate with bidirectional port"""
+        # Create a bidirectional port
+        port_obj = Port(type="bidir", pins=["1", "2"], port_name="test_bidir",
+                       direction=io.Direction.Bidir, options={"all_have_oe": False})
+        platform_port = SiliconPlatformPort("comp", "test_bidir", port_obj)
+
+        # Create buffer
+        buffer = IOBuffer("io", platform_port)
+
+        # Create module and elaborate
+        m = Module()
+        m.submodules.buffer = buffer
+
+        # Get the fragment
+        fragment = Fragment.get(m, None)
+
+        # Just check that elaboration succeeds without error
+        self.assertIsNotNone(fragment)
+
+    def test_elaborate_invert(self):
+        """Test IOBuffer elaborate with inverted port"""
+        # Create an input port with invert=True
+        port_obj = Port(type="input", pins=["1", "2"], port_name="test_input",
+                       direction=io.Direction.Input, options={})
+        platform_port = SiliconPlatformPort("comp", "test_input", port_obj, invert=True)
+
+        # Create buffer
+        buffer = IOBuffer("i", platform_port)
+
+        # Create module and elaborate
+        m = Module()
+        m.submodules.buffer = buffer
+
+        # Get the fragment
+        fragment = Fragment.get(m, None)
+
+        # Just check that elaboration succeeds without error
+        self.assertIsNotNone(fragment)
+
+
+class FFBufferTestCase(unittest.TestCase):
```

Comment:
once again, if you want to test this, a simulation would be a much cleaner way of testing it than all this cruft